main.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"encoding/base64"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/ONSdigital/rm-survey-service/models"
+	"github.com/cloudfoundry-community/go-cfenv"
+	"github.com/gorilla/handlers"
+	"github.com/gorilla/mux"
+)
+
+const (
+	realm       = "sdc"
+	serviceName = "surveysvc"
+	timeFormat  = "2006-01-02T15:04:05Z0700"
+)
+
+var logger *zap.Logger
+
+func init() {
+	logger, _ = zap.NewProduction()
+	defer logger.Sync()
+}
+
+func main() {
+	dataSource, port := configureEnvironment()
+	db := models.InitDB(dataSource)
+
+	// Set up the signal handler to watch for SIGTERM and SIGINT signals so we
+	// can at least attempt to gracefully shut down before the PaaS/docker etc
+	// running us unceremoneously kills us with a SIGKILL.
+
+	api, _ := models.NewAPI(db)
+
+	// Webserver - strictslash set to true to match trailing slashes to routes
+	r := mux.NewRouter().StrictSlash(true)
+	r.HandleFunc("/info", api.Info).Methods("GET")
+	r.HandleFunc("/surveys", use(api.AllSurveys, basicAuth)).Methods("GET")
+	r.HandleFunc("/surveys/{surveyId}", use(api.GetSurvey, basicAuth)).Methods("GET")
+	r.HandleFunc("/surveys/shortname/{shortName}", use(api.GetSurveyByShortName, basicAuth)).Methods("GET")
+	r.HandleFunc("/surveys/ref/{ref}", use(api.GetSurveyByReference, basicAuth)).Methods("GET")
+	r.HandleFunc("/surveys/{surveyId}/classifiertypeselectors", use(api.AllClassifierTypeSelectors, basicAuth)).Methods("GET")
+	r.HandleFunc("/surveys/{surveyId}/classifiertypeselectors/{classifierTypeSelectorId}", use(api.GetClassifierTypeSelectorById, basicAuth)).Methods("GET")
+	http.Handle("/", r)
+
+	// CompressHandler gzips responses where possible
+	compressHandler := handlers.CompressHandler(r)
+	log.Print(http.ListenAndServe(fmt.Sprintf(":%s", port), compressHandler))
+}
+
+func use(h http.HandlerFunc, middleware ...func(http.HandlerFunc) http.HandlerFunc) http.HandlerFunc {
+	for _, m := range middleware {
+		h = m(h)
+	}
+	return h
+}
+
+func basicAuth(h http.HandlerFunc) http.HandlerFunc {
+	// Taken from https://gist.github.com/elithrar/9146306
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		w.Header().Set("WWW-Authenticate", `Basic realm="Restricted"`)
+
+		s := strings.SplitN(r.Header.Get("Authorization"), " ", 2)
+		if len(s) != 2 {
+			http.Error(w, "Not authorized", 401)
```

Comment:
When using http status codes, should always use the constants from the `http` package - e.g. instead of `401` use `http.StatusUnauthorized`